### PR TITLE
Pull over deploy script changes from react.london

### PR DESCRIPTION
**Pull request by lpil**
planefail

### CODE REVIEW

**Comment by asavin** [comment]
@lpil maybe couple of words on motivation of this PR?

**Comment by lpil** [comment]
Sure. Spoke with Chris but should have let notes. I'll annotate the changes.

File: bin/deploy.sh [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
 
+if [ -z "$ENV" ] || [ -z "$RELEASE_TAG" ]
+then
+  echo Usage:
+  echo "  deploy.sh ENVIRONMENT"
+  exit 1
+fi
+
+set -eu
+set -o pipefail
```
Comment (lpil) [inline bin/deploy.sh]: Safer bash options. Set after the variable checking as otherwise we get an ugly error message when the user forgets to pass the argument rather than the friendlier one above.

File: bin/deploy.sh [hunk 2]
```diff
@@ -1,33 +1,34 @@
-#!/bin/bash
-
-set -e
+#!/usr/bin/env bash
 
 ENV=$1
-RELEASE_TAG=$2
+RELEASE_TAG=$(git rev-parse HEAD)
```
Comment (lpil) [inline bin/deploy.sh]: Removes the need for manual setting of a version. The AWS EB version GUI has dates and what env a version is deployed to so I've found that and a SHA to identify commits is sufficient.

This is a matter of taste though!

File: bin/deploy.sh [hunk 3]
```diff
@@ -1,33 +1,34 @@
...
-else
-  VERSION=$ENV
-  docker build -t $APP_NAME:$ENV .
-  docker tag $APP_NAME:$ENV $ECR_REPO:$ENV
-  docker push $ECR_REPO:$ENV
-fi
+VERSION=$RELEASE_TAG
+docker build -t $APP_NAME:$RELEASE_TAG .
+docker tag $APP_NAME:$RELEASE_TAG $ECR_REPO:$RELEASE_TAG
+docker push $ECR_REPO:$RELEASE_TAG
```
Comment (lpil) [inline bin/deploy.sh]: Removed the different path for production and staging for a few reasons.

* Avoids errors from trying to overwrite a tag. There's a -f arg but Chris said it's deprecated.
* It allows us to move a version directly from staging to production, which seems like a very desirable feature as it means we can ensure that the version tested on staging is the same as what we put live.

File: bin/deploy.sh [hunk 4]
```diff
@@ -1,33 +1,34 @@
...
 
 # Apply docker image path to Dockerrun.aws.json template
+cp Dockerrun.aws.json.template Dockerrun.aws.json
```
Comment (lpil) [inline bin/deploy.sh]: Moving the compiled Dockerrun.aws.json out of the repo so the deployer doesn't need to undo the changes to the file after each deploy. Removes risk of accidentally committing the edited version.

File: bin/deploy.sh [hunk 5]
```diff
@@ -36,14 +37,18 @@ ZIP_FILE=$VERSION.zip
...
 # Update the environment to use the new application version
-aws elasticbeanstalk update-environment --environment-name $APP_NAME-$ENV \
-    --version-label $VERSION --region $AWS_REGION
+aws elasticbeanstalk update-environment \
+  --environment-name $APP_NAME-$ENV \
+  --version-label $VERSION \
+  --region $AWS_REGION
+
+echo
+echo Done! AWS EB deployment rollout in progress.
```
Comment (lpil) [inline bin/deploy.sh]: Just formatting for readability and a friendly success message since the previous version just dumped a load of json to the terminal which is not obviously a success.

File: bin/deploy.sh (hunk 3 above)
Comment (lpil) [inline bin/deploy.sh]: TBH I feel like the production deploy should always be moving an existing image from staging (or storage) to production rather than compiling a fresh version.

**Comment by lpil** [comment]
Thanks
